include/utility/dbus_utility.hpp

```diff
@@ -58,6 +58,100 @@ inline types::MapperGetObject getObjectMap(const std::string& objectPath,
     return getObjectMap;
 }
 
+/**
+ * @brief An API to get object, services, and interfaces for an interface.
+ *
+ * The API returns the map of object, services and interfaces in the
+ * subtree that implement a certain interface. If no interfaces are provided
+ * then all the objects, services and interfaces under the subtree will
+ * be returned.
+ *
+ * Note: It will be caller's responsibility to check for empty vector returned
+ * and generate appropriate error.
+ *
+ * @param[in] i_root - path to search for an interface.
+ * @param[in] i_depth - Maximum depth of the tree to search.
+ * @param[in] i_interfaces - List of interfaces to search.
+ *
+ * @return - A map of object and its related services and interfaces, if
+ *           success. If failed, empty map.
+ */
+
+inline types::MapperGetSubTree
+    getObjectSubTree(const std::string& i_subTree, const int& i_depth,
+                     const std::vector<std::string>& i_interfaces)
+{
+    types::MapperGetSubTree l_subTreeMap;
+
+    if (i_subTree.empty())
```

Comment:
Should also check if `i_interfaces` is empty.

Reply:
It can be empty. As mentioned in the doxygen comments that if `i_interfaces` is empty API will return all the objects, services and interfaces under the subtree path.